### PR TITLE
[StructuralMechanicsApplication] Reducing small bottleneck on lumped damping matrix

### PR DESCRIPTION
**Pull request by loumalouomega**
This is a small bottleneck, the mass matrix is created and destroyed, so better directly assign the values

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 1]
```diff
@@ -1846,13 +1846,10 @@ void BaseSolidElement::CalculateDampingMatrixWithLumpedMass(
...
     // 2.-Calculate mass matrix:
     if (alpha > std::numeric_limits<double>::epsilon()) {
-        MatrixType mass_matrix = ZeroMatrix( mat_size, mat_size );
         VectorType temp_vector(mat_size);
         CalculateLumpedMassVector(temp_vector);
         for (IndexType i = 0; i < mat_size; ++i)
-            mass_matrix(i, i) = temp_vector[i];
-
-        noalias( rDampingMatrix ) += alpha * mass_matrix;
+            rDampingMatrix(i, i) += alpha * temp_vector[i];
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:1855]: Shouldn’t you also initialize the dampingmatrix to zero first?

Note to myself, we should also port this to be used by the other elements in the same fashion

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:1855]: It is initialized some lines up

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:1855]: https://github.com/KratosMultiphysics/Kratos/blob/69c59dd82bed1c0f37ee5b177e6e78b9c4ea20ac/applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp#L1829

**Comment by loumalouomega** [comment]
I  promise not to touch more this PR

**Review by frastellini — APPROVED** [review]
Thank you @loumalouomega  for the enhancement!

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 2]
```diff
@@ -1660,25 +1659,29 @@ double BaseSolidElement::CalculateDerivativesOnCurrentConfiguration(
...
-Vector BaseSolidElement::GetBodyForce(
+array_1d<double, 3> BaseSolidElement::GetBodyForce(
     const GeometryType::IntegrationPointsArrayType& IntegrationPoints,
     const IndexType PointNumber
     ) const
 {
-    Vector body_force = ZeroVector(3);
+    array_1d<double, 3> body_force;
+    for (IndexType i = 0; i < 3; ++i)
+        body_force[i] = 0.0;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:1669]: isn't this the default?

File: applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp [hunk 3]
```diff
@@ -108,8 +108,9 @@ void SmallDisplacement::CalculateAll(
 {
     KRATOS_TRY;
 
-    const SizeType number_of_nodes = GetGeometry().size();
-    const SizeType dimension = GetGeometry().WorkingSpaceDimension();
+    auto& r_geometry = this->GetGeometry();
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp:111]: why not const?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/small_displacement.cpp:111]: Not compiled :P

**Pull request merged**